curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -830,6 +840,12 @@ public boolean isZk34CompatibilityMode()
         return zk34CompatibilityMode;
     }
 
+    @Override
+    public boolean isZk35CompatibilityMode()
+    {
+        return !zk34CompatibilityMode && !Compatibility.hasPersistentWatchers();
```

Comment:
BTW why do we need zk34CompatibilityMode as insatnce field? Couldn't we call Compatibility.isZK34() instead?

Reply:
I forget TBH. As I recall, I wanted users to be able to force 3.4 mode regardless of the library. So, someone could use ZK 3.5 client library against a 3.4 server. But this likely isn't important anymore. We should probably consider removing 3.4 support altogether at some point.